### PR TITLE
Add pivot tests

### PR DESCRIPTION
**Pull request by naterush**
# Description

1. Adds a bunch of tests for pivot
2. Adds some useful utilities for checking column headers: checkColumnExists, checkColumnCount, createPivotFromSelectedSheet

# Testing

Run tests locally to make sure they pass there too.

# Documentation

No.

### CODE REVIEW

File: tests/streamlit_ui_tests/features/pivot.spec.ts [hunk 1]
```diff
@@ -1,36 +1,214 @@
...
+        const mito = await getMitoFrameWithTestCSV(page);
+
+        // TODO
+    })
+
+    test('Can edit filters to pivot table', async ({ page }) => {
+        const mito = await getMitoFrameWithTestCSV(page);
+
+        // TODO
+    })
```
Comment (naterush) [inline tests/streamlit_ui_tests/features/pivot.spec.ts:200]: TODO

**Review by naterush — COMMENTED** [review]
Finish the final two tests

File: tests/streamlit_ui_tests/features/set_column_formula_tests.spec.ts [hunk 2]
```diff
@@ -196,6 +201,7 @@ test('No formula around rolling range displays [object Object]', async ({ page }
     await page.keyboard.press('Shift+ArrowLeft');
 
     await mito.locator('#cell-editor-input').press('Enter');
+    await page.waitForTimeout(1000);
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/features/set_column_formula_tests.spec.ts:204]: Hopefully with the new awaitResponse we can get rid of these now?

File: tests/streamlit_ui_tests/utils.ts [hunk 3]
```diff
@@ -176,7 +194,9 @@ export const setFormulaUsingCellEditor = async (
     }
 
     await mito.getByRole('textbox').fill(formula);
+    await awaitResponse(page);
     await mito.locator('#cell-editor-input').press('Enter');
+    await page.waitForTimeout(1000);
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/utils.ts:199]: And this one

**Pull request merged**